server/sonar-server/src/test/java/org/sonar/server/qualitygate/RegisterQualityGatesTest.java

```diff
@@ -52,32 +63,165 @@
 
   @Rule
   public DbTester db = DbTester.create(System2.INSTANCE);
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  private static final String A_RATING_VALUE = Integer.toString(RatingGrid.Rating.A.getIndex());
+  private static final int LEAK_PERIOD = 1;
+  private static final String BUILT_IN_NAME = "Sonar way";
+
+  private DbClient dbClient = db.getDbClient();
+  private DbSession dbSession = db.getSession();
+
+  private QualityGateDao qualityGateDao = dbClient.qualityGateDao();
+  private QualityGateConditionDao gateConditionDao = dbClient.gateConditionDao();
+  private MetricDao metricDao = dbClient.metricDao();
+  private QualityGateConditionsUpdater qualityGateConditionsUpdater = new QualityGateConditionsUpdater(dbClient);
 
-  DbClient dbClient = db.getDbClient();
-  DbSession dbSession = db.getSession();
+  private QualityGates qualityGates = mock(QualityGates.class);
+  private RegisterQualityGates underTest = new RegisterQualityGates(dbClient, qualityGateConditionsUpdater, qualityGates);
 
-  QualityGates qualityGates = mock(QualityGates.class);
-  RegisterQualityGates task = new RegisterQualityGates(dbClient,
-    new QualityGateConditionsUpdater(dbClient),
-    dbClient.loadedTemplateDao(),
-    qualityGates);
+  @Before
+  public void setup() {
+    insertMetrics();
+  }
+
+  @After
+  public void after() {
+    underTest.stop();
+    db.executeUpdateSql("delete from metrics");
```

Comment:
Why are you removing metrics at the end ? It's not already done by DbTester ?

Reply:
I though it was not done, but I was wrong.